### PR TITLE
update dependencies and fixes for TypeScript 4.5

### PR DESCRIPTION
**Pull request by ksjogo**
package.json and package-lock.json had relatively old TS versions which can cause troubles when some higher ts dep is in the embedding repo somehow.
TS switched some declaration types to possibly undefined.
I throw there for now.
There might be better handling possible.

### CODE REVIEW

File: parse_node/parse_import_declaration.ts [hunk 1]
```diff
@@ -46,6 +46,12 @@ export const getImportResPathForEnum = (
 
   const enumDeclarations = enumSymbol.declarations
 
+  if (!enumDeclarations) {
+    throw new Error(
+      `No Enum declartion given`
```
Comment (johnfn) [inline parse_node/parse_import_declaration.ts:51]: Hm, were you running into this issue in practice? Or is this just to satisfy the typechecker?

File: parse_node/parse_property_access_expression.ts [hunk 2]
```diff
@@ -46,6 +46,9 @@ export const parsePropertyAccessExpression = (
   if (isEnumType(exprType)) {
     const symbol = exprType.getSymbol()!
     const declarations = symbol.declarations
+    if (!declarations) {
+      throw new Error('no declaration given')
```
Comment (johnfn) [inline parse_node/parse_property_access_expression.ts:50]: Same question about this one.

**Review by johnfn — APPROVED** [review]

**Comment by johnfn** [comment]
Technically speaking the right way to add an error so that ts2gd handles it properly is like so:

```
      props.addError({
        error: ErrorName. [put a good name for the error here]
        location: node,
        description: `
Declarations are undefined. This is an internal ts2gd bug. Please report it. 
        `,
      })
```

File: parse_node/parse_import_declaration.ts (hunk 1 above)
Comment (ksjogo) [reply parse_node/parse_import_declaration.ts:51]: TS changed the declartions to being optional, so just satisfying the checker here.

File: parse_node/parse_property_access_expression.ts (hunk 2 above)
Comment (ksjogo) [reply parse_node/parse_property_access_expression.ts:50]: As above.

**Comment by ksjogo** [comment]
Will add the more explicit error message.

**Comment by ksjogo** [comment]
Did update with error messages.

**Comment by johnfn** [comment]
Perfect, thank you!

**Pull request merged**